classes/Repository/GanalyticsDataRepository.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * 2007-2020 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\Ps_Googleanalytics\Repository;
+
+use Db;
+
+class GanalyticsDataRepository
+{
+    const TABLE_NAME = 'ganalytics_data';
+
+    /**
+     * findByCartId
+     *
+     * @param int $cartId
+     * @param int $shopId
+     *
+     * @return mixed
+     */
+    public function findDataByCartIdAndShopId($cartId, $shopId)
+    {
+        return Db::getInstance()->getValue(
+            'SELECT data
+            FROM `' . _DB_PREFIX_ . self::TABLE_NAME . '`
+            WHERE id_cart = ' . (int) $cartId . '
+                AND id_shop = ' . (int) $shopId
+        );
+    }
+
+    /**
+     * addNewRow
+     *
+     * @param int $cartId
+     * @param int $shopId
+     * @param string $data
+     *
+     * @return bool
+     */
+    public function addNewRow($cartId, $shopId, $data)
+    {
+        return Db::getInstance()->Execute(
+            'INSERT INTO `' . _DB_PREFIX_ . self::TABLE_NAME . '` (id_cart, id_shop, data)
+            VALUES(\'' . (int) $cartId . '\',\'' . (int) $shopId . '\',\'' . pSQL($data) . '\')
+            ON DUPLICATE KEY UPDATE data = \'' . pSQL($data) . '\';'
```

Comment:
Do we know all the data that we put here ? 🤔 There are no "sensitive" data in this ? I'm thinking about GDPR constraints

Reply:
I don't know... To be confirmed by @colinegin  ?

Reply:
Hello guys,

Sorry but I have no clue, we would need to investigate what data are used.
But I know there has been a few issues related to GDPR on Google Analytics :
- https://github.com/PrestaShop/ps_googleanalytics/issues/46
- https://github.com/PrestaShop/PrestaShop/issues/13911

I hope this can help.

Reply:
I checked the data and there is no personnal information in the table. 

**Table : ps_ganalytics**
![image](https://user-images.githubusercontent.com/35840830/85119052-7ed87000-b221-11ea-9312-2b9b3eaf87ab.png)

**Table : ps_ganalytics_data**
![image](https://user-images.githubusercontent.com/35840830/85119083-8ac43200-b221-11ea-85d8-d83820e985c6.png)

Seems ok no ?